### PR TITLE
cleanup: remove redundant error plumbing

### LINKED ISSUE #6479: p2p: remove redundant error returns in (NodeKey).SaveAs
This code https://github.com/tendermint/tendermint/blob/a9e231b54d9a4ac28ef69f590b9a124108be3169/p2p/key.go#L36-L40 is quite redundant

It can simply be made

```go
return ioutil.WriteFile(filePath, jsonBytes, 0600)
```

thus the entire function looking like this

```go
func (nodeKey NodeKey) SaveAs(filePath string) error {
     jsonBytes, err := tmjson.Marshal(nodeKey)
     if err != nil {
          return err
     }
     return ioutil.WriteFile(filePath, jsonBytes, 0600)
}
```

### PR DESCRIPTION
**Pull request by creachadair**
Fixes #6479 and related cases.

This is a mostly-automated fixup using Comby (https://comby.dev) to reduce lexically-obvious redundant error checks. No functional changes are intended.

Slightly ironically, the original presenting case from #6479 is now gone, but the pattern remains elsewhere in the repository.

### CODE REVIEW

**Comment by creachadair** [comment]
As far as I can tell, the e2e test failure is unrelated to this change.

**Comment by cmwaters** [comment]
Yup it's unrelated (caused from a node having not having any peers to block sync)

File: CHANGELOG_PENDING.md [hunk 1]
```diff
@@ -146,6 +147,7 @@ Friendly reminder: We have a [bug bounty program](https://hackerone.com/tendermi
 - [state/privval] \#6578 No GetPubKey retry beyond the proposal/voting window (@JayT106)
 - [rpc] \#6615 Add TotalGasUsed to block_results response (@crypto-facs)
 - [cmd/tendermint/commands] \#6623 replace `$HOME/.some/test/dir` with `t.TempDir` (@tanyabouman)
+- [cleanup] \#6778 Remove redundant error plumbing
```
Comment (tychoish) [inline CHANGELOG_PENDING.md:150]: I might be more of a rebel about this, but I sort of feel like changelogs shouldn't have things which don't impact interfaces or semantics just tend to hide things that users would care more about, and in that spirit, I'd omit this.

**Comment by creachadair** [comment]
> Yup it's unrelated (caused from a node having not having any peers to block sync)

Is there something I can or should do to give it what it needs to run to completion? Or is this related to the ongoing e2e cleanup work?

File: CHANGELOG_PENDING.md (hunk 1 above)
Comment (creachadair) [reply CHANGELOG_PENDING.md]: I agree. I included it because it was listed in the guidelines, but I don't think it adds any real value. Reverted.

**Comment by cmwaters** [comment]
> Is there something I can or should do to give it what it needs to run to completion? Or is this related to the ongoing e2e cleanup work?

More related to ongoing work so I wouldn't worry too much unless it starts becoming more flaky

**Review by tychoish — APPROVED** [review]

**Review by cmwaters — APPROVED** [review]
👍

**Pull request merged**